src/util/serialise.js

```diff
@@ -43,6 +43,28 @@ export async function serialise (model, obj = {}, method = 'POST') {
         // Guess relationship type if not provided
         if (typeof obj[prop].type === 'undefined') obj[prop].type = plural(camel(prop))
         data.relationships[prop] = { data: Object.assign(obj[prop]) }
+      } else if (
+        obj[prop] !== null &&
+        obj[prop].constructor === Array
+      ) {
+        // validate whole array
+        const ptype = plural(camel(prop))
+        for (var i = 0; i < obj[prop].length; i++) {
```

Comment:
Can this `for` loop be replaced with a `forEach`:

```js
obj[prop].forEach(item => {
  if (typeof item.id === 'string')
})
```

Reply:
I want to mutate the array element (writing the type, in line 54 below) in the loop, is it good practice to do that inside a forEach loop?

Reply:
Shouldn't review things when I'm half asleep :laughing:

Should be a `map` instead of `forEach` in this case, Can leave it as it is now - i'll do it while cleaning up all the utility functions 👍